lms/services/assignment.py

```diff
@@ -82,11 +87,21 @@ def upsert_assignment(
             )
             self._db.add(assignment)
 
+            # For new assignments check if we are copying
+            # from an existing one on the current launch
+            historical_assignment = self.get_copied_from_assignment(lti_params)
+            if (
+                historical_assignment
+                and historical_assignment.extra.get("group_set_id")
+                and not extra.get("group_set_id")
```

Comment:
If we get passed one value directly, use that one. I reckon that the least surprising behavior.